src/LanguageServer/ExternalAccess/Copilot/Handler/RequestContext.cs

```diff
@@ -0,0 +1,19 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace Microsoft.CodeAnalysis.LanguageServer.ExternalAccess.Copilot;
+
+internal readonly struct RequestContext(LspRequestContext context)
+{
+    /// <inheritdoc cref="LspRequestContext.Workspace"/>
+    internal Workspace? Workspace => context.Workspace;
+    /// <inheritdoc cref="LspRequestContext.Solution"/>
+    internal Solution? Solution => context.Solution;
+    /// <inheritdoc cref="LspRequestContext.Document"/>
+    internal Document? Document => context.Document;
+    /// <inheritdoc cref="LspRequestContext.GetRequiredDocument()"/>
+    internal Document GetRequiredDocument() => context.GetRequiredDocument();
+
+    internal T GetRequiredService<T>() where T : class => context.GetRequiredService<T>();
```

Comment:
This feels very broad in exposing everything

Reply:
not necessarily opposed to this - they'd still be limited to accessing services that are directly made available by external access (since they have to pass the generic type to this call)

Reply:
but they have access to the solution.  So they can do `solution.Services.GetService<T>`.  So i don't see why we need to broaden the surface area here.  We should make EA as minimal as possible.  only what is absolutley necessary to expose.

Reply:
With this we can MEF export services and use them in the language server extension itself, right (like [this](https://github.com/333fred/compiler-developer-sdk/blob/2ac004d8772b4f3f6b75582143b103d4e2abb06e/src/Microsoft.CodeAnalysis.CompilerDeveloperSdk/SyntaxVisualizer/DocumentSyntaxInformation.cs#L61))? I think it provides more flexibility